src/Doctrine/Orm/Filter/AttributeFilter.php

```diff
@@ -0,0 +1,272 @@
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\AbstractContextAwareFilter;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\PropertyHelperTrait;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use Doctrine\ORM\QueryBuilder;
+use Doctrine\Persistence\ManagerRegistry;
+use InvalidArgumentException;
+use Psr\Log\LoggerInterface;
+use Sylius\Component\Attribute\AttributeType\CheckboxAttributeType;
+use Sylius\Component\Attribute\AttributeType\DateAttributeType;
+use Sylius\Component\Attribute\AttributeType\DatetimeAttributeType;
+use Sylius\Component\Attribute\AttributeType\IntegerAttributeType;
+use Sylius\Component\Attribute\AttributeType\PercentAttributeType;
+use Sylius\Component\Attribute\AttributeType\SelectAttributeType;
+use Sylius\Component\Attribute\Model\AttributeInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * Filters the collection by value of given attribute IRI
+ */
+class AttributeFilter extends AbstractContextAwareFilter implements FilterInterface
+{
+    use PropertyHelperTrait;
+
+    private IriConverterInterface $iriConverter;
+
+    public const OPERATOR_EXACT = 'exact';
+
+    public const OPERATOR_PARTIAL = 'partial';
+
+    public const ATTRIBUTE_ID = 'attribute_id';
+
+    public const VALUE = 'value';
+
+    public function __construct(
+        ManagerRegistry $managerRegistry,
+        IriConverterInterface $iriConverter,
+        ?RequestStack $requestStack = null,
+        LoggerInterface $logger = null,
+        array $properties = null,
+        NameConverterInterface $nameConverter = null
+    ) {
+        parent::__construct(
+            $managerRegistry,
+            $requestStack,
+            $logger,
+            $properties,
+            $nameConverter
+        );
+
+        $this->iriConverter = $iriConverter;
+    }
+
+    /**
+     * @param mixed $value
+     */
+    protected function filterProperty(
+        string $property,
+        $value,
+        QueryBuilder $queryBuilder,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $resourceClass,
+        string $operationName = null
+    ): void {
+        if (
+            !\is_array($value) ||
+            !$this->isPropertyEnabled($property, $resourceClass)
+        ) {
+            return;
+        }
+
+        $attributeId = $this->getAttributeId($value, $property);
+        $extractedValue = $this->extractValue($value, $property);
+        if (null === $attributeId || null === $extractedValue) {
+            return;
+        }
+
+        $alias = $queryBuilder->getRootAliases()[0];
+
+        if ($this->isPropertyNested($property . '.id', $resourceClass)) {
+            [$alias] = $this->addJoinsForNestedProperty($property . '.id', $alias, $queryBuilder, $queryNameGenerator, $resourceClass);
+        }
+
+        Assert::string($alias);
```

Comment:
Do we want to Assert here? Maybe an early return would be better?

Reply:
Let's leave the assert as code analysis gets confused after line 96 with the type of $alias. It considers it as string|mixed while it should be just string

Reply:
Ok but as the Assert is throwing an `\InvalidArgumentException`, won't be a problem in runtime?

Reply:
I can cast it while passing this var to method